app/jobs/cangaroo/poll_job.rb

```diff
@@ -1,61 +1,43 @@
 module Cangaroo
-  class PollJob < ActiveJob::Base
-    include Cangaroo::LoggerHelper
-    include Cangaroo::ClassConfiguration
+  class PollJob < BaseJob
+    class_configuration :frequency, 1.day
 
-    queue_as :cangaroo
+    attr_reader :current_time
 
-    class_configuration :connection
-    class_configuration :frequency, 1.day
-    class_configuration :path, ''
-    class_configuration :parameters, {}
+    def transform
+      { last_poll: last_poll_timestamp.to_i }
+    end
 
     def perform(*)
-      current_time = DateTime.now
+      @current_time = DateTime.now
 
       if !perform?(current_time)
         Cangaroo.logger.info 'skipping poll', job_tags
         return
       end
 
-      Cangaroo.logger.info 'initiating poll', job_tags(last_poll: last_poll_timestamp.to_i)
+      restart_flow(connection_request)
```

Comment:
PollJobs should really `GET` on their endpoints, not use POST. Otherwise you need to have something like this in your `routes.rb`:

```
  resources :customers do
    post 'index' => 'customers#index', on: :collection
  end
```